File: C++/Array_HashMap_Medium_CountTheNumberOfBeautifulSubarrays_(CountSubarraysWithGivenXOR_K0)_2588.cpp
class Solution {
public:
    long long beautifulSubarrays(vector<int>& nums)
    {
        cin.tie(nullptr);
        cout.tie(nullptr);
        ios::sync_with_stdio(false);

        long long masterCount = 0;
        unordered_map<int, int> XORMap;
        long long XORSum = 0;

        XORMap[0] = 1;

        for(int i = 0; i < nums.size(); i++)
        {
            XORSum = XORSum ^ nums[i];

            int remove = XORSum ^ 0;

            masterCount += XORMap[remove];

            XORMap[XORSum]++;
        }

        return masterCount;
    }
};

File: Python/Array_HashMap_Medium_CountTheNumberOfBeautifulSubarrays_(CountSubarraysWithGivenXOR_K0)_2588.py
class Solution:
    def beautifulSubarrays(self, nums: List[int]) -> int:
        XORSum = 0
        masterCount = 0
        XORMap = defaultdict(int)

        XORMap[0] = 1

        for i in range(0, len(nums)):

            XORSum = XORSum ^ nums[i]

            remove = XORSum ^ 0

            masterCount += XORMap[remove]

            XORMap[XORSum] += 1

        return masterCount

